create_primers.py

```diff
@@ -0,0 +1,261 @@
+import math
+import re
+import argparse
+import pandas as pd
+from Bio.SeqUtils import MeltingTemp as mt
+from Bio.Seq import Seq
+
+
+def Parser():
+    """Returns command line parser."""
+    parser = argparse.ArgumentParser(
+            description='Script by Caelan Radford, Adam Dingens, and Jesse Bloom to design specified codon tiling primers with specific melting temperatures.',
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            )
+
+    parser.add_argument('sequencefile', help="the name of a file giving the sequence for which we are designing the primers. This file should only contain the sequence, and should not have any headers or other content. For the sequence, make the 5' and 3' ends that you do not want to mutate in lower case. Make the portion of the coding sequence that you want to tile with ambiguous codons in upper case. Typically, for example, the first site you mutate would be the initial start codon, so the first *ATG* would be the first upper case letter. You must have at least *(startprimerlength - 3) / 2* nucleotides in lower case at each end of the upper case sequence that you are mutating. This is because at least this much flanking sequence is needed to design primers of the indicated length; more sequence may be required if the primer at either end is extended beyond the startprimerlength.")
+    parser.add_argument('mutations_csv', help="the name of a csv file containing a table of specific mutations per site you want to make. It should have one column ('site') with site numbers and one column ('mutant') with amino acid mutants to make at that site. Site 1 is the first codon in the uppercase sequence.")
+    parser.add_argument('codon_frequency_csv', help="the name of a csv file with a table of codon frequencies to be used to determine which codons to mutate to. It should have one column 'aa' with single letter amino acids, one column 'codon' with codons, and one column 'frequency' with codon frequencies. The highest frequency codon for each amino acid is used.")
+    parser.add_argument('primerprefix', help="prefix name to be added to each primer")
+    parser.add_argument('outfile', help='name of primer output file')
+    parser.add_argument('--startprimerlength', type=int, help='starting primer length', default=37)
+    parser.add_argument('--maxprimertm', type=float, help="Upper temperature limit for primers.", default=61)
+    parser.add_argument('--minprimertm', type=float, help="Lower temperature limit for primers.", default=60)
+    parser.add_argument('--minlength', type=int, help='Minimum primer length', default=25)
+    parser.add_argument('--maxlength', type=int, help='Maximum primer length', default=51)
+    return parser
+
+
+def ReverseComplement(seq):
+    """Returns reverse complement of sequence. Preserves upper/lower case."""
+    d = {'A':'T', 'T':'A', 'C':'G', 'G':'C', 'a':'t', 't':'a', 'c':'g', 'g':'c', 'N':'N', 'n':'n', 'S':'S', 's':'s', 'K':'M', 'k':'m'}
```

Comment:
Is there a reason my `S`, `K`, and `M` are the only ambiguous nucleotides included?

Reply:
This was a copy over from the old script. We could add more, which ones do you want to add?

Reply:
I guess my main point is that this should be deliberate and not just a copy/paste, so it should include what you think is necessary for the script (or maybe all of the degenerate nucleotides if you think at one point this script might allow degenerate nucleotide functionality).